plugins/ci/pkg/ci/tfsec.go

```diff
@@ -10,92 +10,98 @@ import (
 
 	"github.com/fairwindsops/insights-plugins/plugins/ci/pkg/commands"
 	"github.com/fairwindsops/insights-plugins/plugins/ci/pkg/models"
+	"github.com/hashicorp/go-multierror"
 	"github.com/sirupsen/logrus"
 )
 
 func (ci *CIScan) TerraformEnabled() bool {
 	return *ci.config.Reports.TFSec.Enabled
 }
 
-func (ci *CIScan) ProcessTerraformPaths() (report models.ReportInfo, areResults bool, err error) {
+func (ci *CIScan) ProcessTerraformPaths() (report *models.ReportInfo, errs error) {
 	logrus.Infof("processing %d Terraform paths", len(ci.config.Terraform.Paths))
 	if len(ci.config.Terraform.Paths) == 0 {
-		return models.ReportInfo{}, false, nil
+		return nil, nil
 	}
+	allErrs := new(multierror.Error)
 	var reportProperties models.TFSecReportProperties
+	TFSecVersion, err := commands.Exec("tfsec", "-v")
+	if err != nil {
+		return nil, fmt.Errorf("cannot get the version of tfsec: %v: %s", err, TFSecVersion)
+	}
+	TFSecVersion = strings.TrimPrefix(TFSecVersion, "v")
 	for _, terraformPath := range ci.config.Terraform.Paths {
 		results, err := ci.ProcessTerraformPath(terraformPath)
 		if err != nil {
-			return models.ReportInfo{}, false, err
+			allErrs = multierror.Append(allErrs, fmt.Errorf("unable to process path %q: %v", terraformPath, err))
 		}
 		if len(results) > 0 {
 			reportProperties.Items = append(reportProperties.Items, results...)
 		}
 	}
+	var allErrsCombined error = nil
+	if len(allErrs.Errors) > 0 { // keep the multierror from becoming individual action items
+		allErrsCombined = fmt.Errorf("%v", allErrs.Error()) // return string representation from the multierror
```

Comment:
In some other places, you have used `allErrs.ErrorOrNil()`, wasn't possible to do it here?

Reply:
ohh, nvm I think I got it